docs/src/api/class-browsercontext.md

```diff
@@ -155,6 +155,35 @@ await browser_context.add_cookies([cookie_object1, cookie_object2])
 browser_context.add_cookies([cookie_object1, cookie_object2])
 ```
 
+## event: BrowserContext.request
+- argument: <[Request]>
+
+Emitted when a request is issued from any pages created through this context.
+The [request] object is read-only. In order to intercept and mutate requests,
+see [`method: BrowserContext.route`] or [`method: page.route`].
+
+## event: BrowserContext.requestFailed
+- argument: <[Request]>
+
+Emitted when a request fails, for example by timing out.
```

Comment:
Let's also reference corresponding page events from these ones, and vice versa. Something like `to only listen for requests from a particular page, use [...page.requestFailed reference...]`.